storage-proofs/post/src/fallback/vanilla.rs

```diff
@@ -562,102 +600,303 @@ mod tests {
+
+        let leaves = 64 * get_base_tree_count::<Tree>();
+        let sector_size = leaves * NODE_SIZE;
+
+        let pub_params = PublicParams {
+            sector_size: sector_size as u64,
+            challenge_count: 10,
+            sector_count,
+        };
+
+        let randomness = <Tree::Hasher as Hasher>::Domain::random(rng);
+        let prover_id = <Tree::Hasher as Hasher>::Domain::random(rng);
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let temp_path = temp_dir.path();
+
+        let mut pub_sectors = Vec::new();
+        let mut priv_sectors = Vec::new();
+
+        let mut trees = Vec::new();
+
+        let mut faulty_sectors = Vec::<SectorId>::new();
+
+        for _i in 0..total_sector_count {
+            let (_data, tree) =
+                generate_tree::<Tree, _>(rng, leaves, Some(temp_path.to_path_buf()));
+            trees.push(tree);
+        }
+
+        let faulty_denominator = 3;
+
+        let (_data, wrong_tree) =
+            generate_tree::<Tree, _>(rng, leaves, Some(temp_path.to_path_buf()));
+
+        for (i, tree) in trees.iter().enumerate() {
+            let make_faulty = i % faulty_denominator == 0;
+
+            let comm_c = <Tree::Hasher as Hasher>::Domain::random(rng);
+            let comm_r_last = tree.root();
+
+            priv_sectors.push(PrivateSector {
+                tree: if make_faulty { &wrong_tree } else { tree },
+                comm_c,
+                comm_r_last,
+            });
+
+            let comm_r = <Tree::Hasher as Hasher>::Function::hash2(&comm_c, &comm_r_last);
+
+            if make_faulty {
+                faulty_sectors.push((i as u64).into());
+            }
+
+            pub_sectors.push(PublicSector {
+                id: (i as u64).into(),
+                comm_r,
+            });
+        }
+
+        let pub_inputs = PublicInputs {
+            randomness,
+            prover_id,
+            sectors: &pub_sectors,
+            k: None,
+        };
+
+        let priv_inputs = PrivateInputs::<Tree> {
+            sectors: &priv_sectors[..],
+        };
+
+        let proof = FallbackPoSt::<Tree>::prove_all_partitions(
+            &pub_params,
+            &pub_inputs,
+            &priv_inputs,
+            partitions,
+        );
+
+        match proof {
+            Ok(proof) => {
+                let is_valid =
+                    FallbackPoSt::<Tree>::verify_all_partitions(&pub_params, &pub_inputs, &proof)
+                        .expect("verification failed");
+                assert!(!is_valid, "PoSt returned a valid proof with invalid input");
+            }
+            Err(e) => match e.downcast::<Error>() {
+                Err(_) => panic!("failed to downcast to Error"),
+                Ok(Error::FaultySectors(sector_ids)) => assert_eq!(faulty_sectors, sector_ids),
+                Ok(_) => panic!("PoSt failed to return FaultySectors error."),
+                _ => (),
```

Comment:
Shouldn't all cases be already covered by `Err(_)` and `Ok(`_`)?

Reply:
Yes, that's vestigial, thanks.